File: problemsets/Codeforces/C++/A844.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include <bits/stdc++.h>
using namespace std;
int main() {
  string a; cin >> a;
  set<char> u;
  for (char &c: a) u.insert(c);
  int n = a.size(), k, m = u.size(); cin >> k;
  if (k > n) cout << "impossible";
  else cout << max(0, k-m);
}

File: problemsets/Codeforces/Python/A844.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
a=input()
n=len(a)
k=int(input())
print([max(0,k-len({*a})),'impossible'][n<k])